File: mini-claw-code-ts/src/mock.ts
import type { AssistantTurn, Message, Provider, ToolDefinition } from "./types";

export class MockProvider implements Provider {
  #responses: AssistantTurn[];
  #cursor = 0;

  constructor(responses: Iterable<AssistantTurn>) {
    this.#responses = [...responses];
  }

  async chat(_messages: Message[], _tools: ToolDefinition[]): Promise<AssistantTurn> {
    const turn = this.#responses[this.#cursor];
    if (!turn) {
      throw new Error("MockProvider: no more responses");
    }

    this.#cursor += 1;
    return structuredClone(turn);
  }
}

File: mini-claw-code-py/src/mini_claw_code_py/mock.py
from __future__ import annotations

from collections import deque
from typing import Sequence

from .types import AssistantTurn, Message, Provider, ToolDefinition


class MockProvider(Provider):
    def __init__(self, responses: deque[AssistantTurn]) -> None:
        self._responses = responses

    async def chat(
        self,
        _messages: Sequence[Message],
        _tools: Sequence[ToolDefinition],
    ) -> AssistantTurn:
        if not self._responses:
            raise RuntimeError("MockProvider: no more responses")
        return self._responses.popleft()
